### ISSUE TITLE
Any way to simulate a server error?

### ISSUE THREAD

**Issue by austinknight** [thoov/mock-socket#126]
I'm wanting to test that something for websocket.onerror, what would I need to do in the mocked server to invoke this on the websocket?

**Comment by thoov** [comment]
Right now the only way to trigger the onerror is by trying to connect to an invalid or closed connection. Something like this:

```js
it('.onerror will be fired', (done) => {
  // uncomment the following line and onerror will not be triggered
  //const server = new Server('ws://localhost:8080');
  const socket = new WebSocket('ws://localhost:8080');

  socket.onerror = function error() {
    assert.ok(true, 'websocket does not have a server to connect to');
    done();
  };
});
```

If you are looking for something on the server to manually trigger an error like so:

```js
it('errors can be sent from the server', (done) => {
  const server = new Server('ws://localhost:8080');
  const socket = new WebSocket('ws://localhost:8080');

  server.on('connection',  () => {
    server.triggerError() // this does not exist right now
  })

  socket.onerror = function error() {
    assert.ok(true, 'the server has sent an error');
    done();
  };
});
```

I can add the functionality. Just let me know.

**Comment by austinknight** [comment]
Yes, the second example is what I am talking about. Adding a triggerError method on the server mock would be awesome!

**Comment by stevenxxiu** [comment]
@thoov I don't think the first is triggered even when using websockets instead of socket io. A way to do this would be great!

**Comment by TimMurnaghan** [comment]
@thoov I guess that triggerError never happened?
Sorry to add a "me too", but this is an important function of a mock library. I want to exercise the error handling of my client code (which, given the way developers think, is often broken). And it's definitely your second case. I don't want an invalid endpoint - it should be valid - but causing errors in some cases.

Should I do it and make a pull request?  Or is it big enough that it will cause conflicts with your refactor?

**Comment by thoov** [comment]
@TimMurnaghan Thanks for reminding me. Just published 7.1.0 where you can call `server.simulate('error');`. Here is the PR: #184
